src/jv_parse.c

```diff
@@ -703,11 +702,7 @@ static pfunc scan_json(struct jv_parser* p, char ch, jv* out) {
   presult answer = 0;
   p->last_ch_was_ws = 0;
   if (p->st == JV_PARSER_NORMAL) {
-    if(ch == '#') {
```

Comment:
So line comments are gone?  Can you squash this with the other commit?

Reply:
Yeap can do that. When i rebased i tried to keep the existing commits as intact as possible as a first step.

Trying to understand the `0` -> `OK` change. `OK` is `static const presult OK = "output produced";` so i guess something else must have changed before the change? only reference i can find is in https://github.com/jqlang/jq/pull/2548#issuecomment-1628024506. Will digg more later.

Reply:
Note to self: Remember to update PR title and commits to reflect if # is support or not

Reply:
I don't actually see that to begin with in the original PR code at https://github.com/liquidaty/jq/blob/strip-comments/src/jv_parse.c, so I'm not sure how it got there to begin with before it was removed-- but maybe I'm confusing myself and I put it in some earlier or later version than I'm now looking at.

In any case, if you decide to keep line comments, then another way to achieve, that possibly reduces redundancy in the code base (assuming you keep `//` and `/*` comment support)-- and at least to me is more readable and easier to maintain-- would be to remove any other code related to `#` comments and then insert:

```diff
static pfunc scan_json(struct jv_parser* p, char ch, jv* out) {
  ...
    if(ch == '/' && (p->flags & JV_PARSE_STRIP_COMMENTS)) {
      p->scan = scan_slash_comment;
      return answer;
    }
+    if(ch == '#' && (p->flags & JV_PARSE_STRIP_COMMENTS)) {
+      p->scan = scan_line_comment;
+      return answer;
+    }
```

Reply:
As it seems i was confused about json5 supporting `#`-comments (it does not) my opinion is that we can skip support them for now